packages/truffle-db/src/workspace/pouch.ts

```diff
@@ -37,29 +48,27 @@ const resources = {
 }
 
 export class Workspace {
-  sources: PouchDB.Database;
-  bytecodes: PouchDB.Database;
-  compilations: PouchDB.Database;
-  contracts: PouchDB.Database;
-  contractInstances: PouchDB.Database;
-  networks: PouchDB.Database;
+
+  private dbApi: PouchApi;
 
   private ready: Promise<void>;
 
   constructor () {
+    this.dbApi = {}
```

Comment:
Ah, I see. Would this work as `this.dbApi;`? We just talked about this via zoom, so just leaving this here for now!

Reply:
That line initializes an object and assigns it to `this.dbAp`. If it weren't initialized, this.dbApi remains `undefined` and we couldn't treat it as an object. Which leads to another problem :( `this.dbApi` is populated dynamically in a loop and I'm not sure TS can properly infer types and will complain.

Reply:
Yep, I'm ok leaving it as is. I sort of liked the extra check to make sure each resource is actually a part of the object but I don't think its really necessary.